recipes/qt/6.x.x/conanfile.py

```diff
@@ -522,26 +526,133 @@ def package_id(self):
                 self.info.settings.compiler.runtime = "MT/MTd"
 
     def package_info(self):
-        # FIXME add components
-        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.names["cmake_find_package_multi"] = "Qt6"
 
-        # Add top level include directory, so code compile if someone uses
-        # includes with prefixes (e.g. "#include <QtCore/QString>")
-        self.cpp_info.includedirs = ["include"]
+        libsuffix = ""
+        if self.settings.build_type == "Debug":
+            if self.settings.os == "Windows":
+                libsuffix = "d"
+            if tools.is_apple_os(self.settings.os):
+                libsuffix = "_debug"
+
+        def _create_module(module, requires=[]):
+            assert module not in self.cpp_info.components, "Module %s already present in self.cpp_info.components" % module
+            self.cpp_info.components[module].names["cmake_find_package"] = module
+            self.cpp_info.components[module].libs = ["Qt6%s%s" % (module, libsuffix)]
+            self.cpp_info.components[module].includedirs = ["include", os.path.join("include", "Qt%s" % module)]
+            self.cpp_info.components[module].defines = ["QT_%s_LIB" % module.upper()]
+            if module != "Core" and "Core" not in requires:
+                requires.append("Core")
+            self.cpp_info.components[module].requires = requires
+
+        def _create_plugin(pluginname, libname, type, requires):
+            assert pluginname not in self.cpp_info.components, "Plugin %s already present in self.cpp_info.components" % pluginname
+            self.cpp_info.components[pluginname].names["cmake_find_package"] = pluginname
+            self.cpp_info.components[pluginname].libs = [libname + libsuffix]
+            self.cpp_info.components[pluginname].libdirs = [os.path.join("res", "archdatadir", "plugins", type)]
+            self.cpp_info.components[pluginname].includedirs = []
+            if "Core" not in requires:
+                requires.append("Core")
```

Comment:
Since the "requires" only contains external deps could we do this?

```suggestion
                requires.insert(0, "Core")
```

Reply:
requires contain external (as in other packages) and internal deps (as in component of the qt recipe), but I'm not sure to understand what the order of these requirements is supposed to change ?

Reply:
I am curious how the link order will be affected. `a -> b -> c -> a` needs to be `b -> c -> a`
I tested it locally and it seems conan adds them in double so there is no problem (stays like the first list)